File: solutions/Cyclemin/solve/java/CycleminSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class CycleminSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        String s = (String) reader.next(String.class);
        reader.next();
        
        int k = (Integer) reader.next(Integer.class);
        reader.close();

        Cyclemin solver = new Cyclemin();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.bestmod(s, k));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}

File: solutions/Cyclemin/solve/ruby/CycleminSolver.rb
#! /usr/bin/env ruby

require_relative "Cyclemin"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    s = reader.next(TString)
    reader.next()
    k = reader.next(TInt)

    result = Cyclemin.new().bestmod(s, k)
    IO.write(ARGV[1], Writer.new().next(result, TString).to_s())
end
init()
main()
